### PR TITLE
Correct class trainer icons.

### PR DESCRIPTION
**Pull request by lineagedr**
## Changes Proposed:
-  Correct the class trainer icons
-  

## Issues Addressed:
- Closes https://github.com/azerothcore/azerothcore-wotlk/issues/4249

## Tests Performed:
-
-

## How to Test the Changes:

Clear wow cache before testing.

## Known Issues and TODO List:

- [ ]
- [ ] 

## Target Branch(es):
- [x] Master

 
## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here in the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

### CODE REVIEW

File: src/server/game/Handlers/QueryHandler.cpp [hunk 1]
```diff
@@ -109,7 +109,15 @@ void WorldSession::HandleCreatureQueryOpcode(WorldPacket& recvData)
         data << Name;
         data << uint8(0) << uint8(0) << uint8(0);           // name2, name3, name4, always empty
         data << Title;
-        data << ci->IconName;                               // "Directions" for guard, string for Icons 2.3.0
+        if (ci->npcflag & UNIT_NPC_FLAG_TRAINER_CLASS)
+        {
+            if (Player* player = GetPlayer())
+                player->getClass() == ci->trainer_class ? data << "Trainer" : data << "Speak";
+            else
+                data << ci->IconName;
```
Comment (Helias) [inline src/server/game/Handlers/QueryHandler.cpp:117]:
```suggestion
            if (Player* player = GetPlayer())
            {
                data << player->getClass() == ci->trainer_class ?  "Trainer" : "Speak";
            }
            else
            {
                data << ci->IconName;
            }
```

File: src/server/game/Handlers/QueryHandler.cpp [hunk 2]
```diff
@@ -109,7 +109,15 @@ void WorldSession::HandleCreatureQueryOpcode(WorldPacket& recvData)
...
+        }
+        else
+            data << ci->IconName;                           // "Directions" for guard, string for Icons 2.3.0
```
Comment (Helias) [inline src/server/game/Handlers/QueryHandler.cpp:120]:
```suggestion
        else
        {
            data << ci->IconName;                           // "Directions" for guard, string for Icons 2.3.0
        }
```

**Review by Helias — CHANGES_REQUESTED** [review]

**Comment by pklloveyou** [comment]
tested

**Comment by Voxstrasza** [comment]
tested, works good.

**Comment by Kitzunu** [comment]
Please fix the build :)

**Comment by Voxstrasza** [comment]
@lineagedr why close?

**Comment by lineagedr** [comment]
@Voxstrasza HandleCreatureQueryOpcode is only sent once per account and then cached in wow. Login with a warrior you'll see all works well. Now login with a different char you'll see that only the Warrior trainer has the spellbook icon. This PR won't fix it.

**Comment by BarbzYHOOL** [comment]
What can be done then?

**Pull request closed without merging**